addons/mail/models/mail_mail.py

```diff
@@ -138,10 +151,13 @@ def process_email_queue(self, ids=None):
                                 messages to send (by default all 'outgoing'
                                 messages are sent).
         """
+        # scheduled_date is in local time in db => user_now has to be in local time for comparison
+        user_tz = pytz.timezone(self.env.context.get('tz') or self.env.user.tz or 'UTC')
+        user_now = user_tz.fromutc(datetime.datetime.now())
```

Comment:
Shouldn't we simply use ``utcnow`` ? As anyway we are comparing utc values.

Reply:
For me no, we are not comparing utc values, because the user is entering the string as scheduled date without thinking about the timezone he is in and then he saves it in db. Using utcnow could lead to misunderstandings, for example the user is setting a scheduled_date in one hour but the mail is directly sent because he didn't think about the time zone he was in and the fact that he had to compute the time difference with utcnow.

Reply:
That's breaking odoo rule: all values should be encoded as UTC. That's the webclient purpose to convert datetimes. But here indeed we have an issue, people may put whatever they want in that field. After that most emails are sent using the cron (aka with root that has no timezone). We don't have any clue what was the source timezone, so we cannot guess anything. Better consider it is UTC and if that was badly encoded then we cannot do anything about it.